### PR TITLE
Fix unbound local variable issue.

### PR DESCRIPTION
**Pull request by idegtiarov**
**Description:**

Fix issue with the unbound local variable in the `api` module.

The issue happens when credit service is not found in the runtime.

I'd like to clarify several points that are also touched on the PR to make collaboration easier:
-  Should I follow OeX `summary line rule` from the [python style guide docstrings section](https://edx.readthedocs.io/projects/edx-developer-guide/en/latest/style_guides/python-guidelines.html#docstrings) or it could be ignored?
- The same question is related to breaking long lines with the help of backslash (I personally would like to avoid that syntax, but maybe it is ok)?
- The library has lots of places with the "old" python2 style, any plan for major refactoring or it could be lightly changed in the  
updated code?

**Pre-Merge Checklist:**

- [x] Updated the version number in `edx_proctoring/__init__.py` and `package.json` if these changes are to be released.
- [x] Described your changes in `CHANGELOG.rst`
- [x] Confirmed Github reports all automated tests/checks are passing.
- [x] Approved by at least one additional reviewer.

**Post-Merge:**

- [ ] Create a tag matching the new version number.

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @idegtiarov! I've created [OSPR-5100](https://openedx.atlassian.net/browse/OSPR-5100) to keep track of it in JIRA.

As a core committer in this repo, you can merge this once the pull request is approved per the [core committer reviewer requirements](https://openedx.atlassian.net/wiki/spaces/COMM/pages/1529675973/Open+edX+Core+Committers#Reviews-and-Ownership) and according to the agreement with your edX Champion.


This is currently a draft pull request. When it is ready for our review and all tests are green, click "Ready for Review", or remove "WIP" from the title, as appropriate.

**Comment by natabene** [comment]
@idegtiarov Thank you for your contribution. Please let me know once it is ready for our review.

File: edx_proctoring/api.py
Comment (idegtiarov) [inline edx_proctoring/api.py:1652]: I've added `pylint disable` here because pylint warning looks like a checker's bug. 

One of the complained issue's thread https://github.com/PyCQA/pylint/issues/741

Comment (schenedx) [reply edx_proctoring/api.py:1652]: What I would do here to avoid the `pylint diable` is to just use more variables to capture the conditions we want to test out. I think use more lines of code and variables make the code easier to read.
```
attempt = get_exam_attempt(exam['id'], user_id)
graded_exam = not exam['is_practice_exam']
due_date_is_passed = credit_state and has_due_date_passed(credit_state.get('course_end_date', None))

if attempt:
    status = attempt['status']
elif graded_exam and due_date_is_passed:
    status = ProctoredExamStudentAttemptStatus.expired
...
```

**Review by schenedx — COMMENTED** [review]
@idegtiarov Thank you for the change. I think it largely looks good. I gave a suggestion for how to make the elif statement easier to read. Let me know what you think.

**Comment by idegtiarov** [comment]
@schenedx Thanks for the review your proposal would work. It is more a workaround for me than a solution but I'm going to follow it. 

Do you have any thoughts about my questions in the PR description?

**Comment by schenedx** [comment]
@idegtiarov 
Q1: I am not dogmatic about that rule. I think what you have in the PR is fine. 
Q3: We currently don't have plan to refactor the repo to get the code style match fully with python 3.

**Comment by idegtiarov** [comment]
Got it thanks!

**Review by schenedx — APPROVED** [review]
👍

**Comment by natabene** [comment]
@schenedx or @idegtiarov Please merge when you feel timing is good.

**Comment by edx-webhook** [comment]
@idegtiarov 🎉 Your pull request was merged!

Please take a moment to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/edx-proctoring&entry.867055334=https://github.com/edx/edx-proctoring/pull/719&entry.1484655318=https://github.com/idegtiarov&entry.752974735=2020-11-02+16:50&entry.1917517419=2020-11-09+15:18&entry.2133058324=Yes) so we can improve your experience in the future.

**Comment by edx-webhook** [comment]
@nedbat, @nasthagiri, @schenedx: thought you might like to know that idegtiarov merged this pull request.

**Pull request merged**
